src/client/pythonEnvironments/common/environmentManagers/activestate.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import * as path from 'path';
+import { pathExists, shellExecute } from '../externalDependencies';
+import { cache } from '../../../common/utils/decorators';
+import { traceError, traceVerbose } from '../../../logging';
+import { isTestExecution } from '../../../common/constants';
+import { getOSType, getUserHomeDir, OSType } from '../../../common/utils/platform';
+
+const STATE_GENERAL_TIMEOUT = 5000;
+
+export type ProjectInfo = {
+    name: string;
+    organization: string;
+    local_checkouts: string[]; // eslint-disable-line camelcase
+    executables: string[];
+};
+
+export async function isActiveStateEnvironment(interpreterPath: string): Promise<boolean> {
+    const execDir = path.dirname(interpreterPath);
+    const runtimeDir = path.dirname(execDir);
+    return pathExists(path.join(runtimeDir, '_runtime_store'));
+}
+
+export class ActiveState {
+    private static statePromise: Promise<ActiveState | undefined> | undefined;
+
+    public static async getState(): Promise<ActiveState | undefined> {
+        if (ActiveState.statePromise === undefined || isTestExecution()) {
+            ActiveState.statePromise = ActiveState.locate();
+        }
+        return ActiveState.statePromise;
+    }
+
+    public static getStateToolDir(): string | undefined {
+        const home = getUserHomeDir();
+        if (!home) {
+            return undefined;
+        }
+        return getOSType() === OSType.Windows
+            ? path.join(home, 'AppData', 'Local', 'ActiveState', 'StateTool')
+            : path.join(home, '.local', 'ActiveState', 'StateTool');
+    }
+
+    private static async locate(): Promise<ActiveState | undefined> {
+        const stateToolDir = this.getStateToolDir();
+        if ((stateToolDir && (await pathExists(stateToolDir))) || isTestExecution()) {
```

Comment:
```suggestion
        if ((stateToolDir && (await pathExists(stateToolDir)))) {
```
Can you mock `home` directory in tests instead? See: https://github.com/microsoft/vscode-python/blob/1ae84e9a5a5f953121e6bdb00b4b6982b58266ea/src/test/pythonEnvironments/common/environmentManagers/conda.unit.test.ts#L106